src/Shared/AWS/AWSSemanticConventions.Alt.Base.cs

```diff
@@ -0,0 +1,180 @@
+
+using OpenTelemetry.SemanticConventions;
+
+
+namespace OpenTelemetry.AWS;
+
+// disable Style Warnings to improve readability of this specific file.
+#pragma warning disable SA1124
+#pragma warning disable SA1005
+#pragma warning disable SA1514
+#pragma warning disable SA1201
+#pragma warning disable SA1623
+
+internal static partial class AWSSemanticConventions
+{
+    /// <summary>
+    /// Defines all Semantic Conventions used by AWS extension projects.
+    ///
+    /// All values default to <c>string.Empty</c> and are then is only defined
+    /// in the first version specific class (ie <see cref="AWSSemanticConventions_v1_10"/>)
+    /// to use it.  This helps ensure the attribute doesn't get used if the user has specified
+    /// a specific <see cref="SemanticConventionVersion"/>.
+    ///
+    /// See <see cref="AWSSemanticConventions.GetSemanticConventionVersion"/> for details.
+    /// </summary>
+    private abstract class AWSSemanticConventionsBase
+    {
+        // CLOUD Attributes
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudAccountId"/>
+        public virtual string AttributeCloudAccountID => string.Empty;
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudAvailabilityZone"/>
+        public virtual string AttributeCloudAvailabilityZone => string.Empty;
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudPlatform"/>
+        public virtual string AttributeCloudPlatform => string.Empty;
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudProvider"/>
+        public virtual string AttributeCloudProvider => string.Empty;
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudRegion"/>
+        public virtual string AttributeCloudRegion => string.Empty;
+        /// <inheritdoc cref="CloudAttributes.AttributeCloudResourceId"/>
+        public virtual string AttributeCloudResourceId => string.Empty;
```

Comment:
I thought about doing that, but decided to leave the Base class as always string.Empty.  It's very important that when a new Attribute is added, the existing Semantic Convention Version classes do not have a value.  Otherwise, it could accidently start emitting newer telemetry.

By leaving the base class Attribute as empty, it will hopefully be more obvious that the value should be `string.Empty` in the base class and the actual value is defined in the first implementation class that uses it.

For example, if a new Attribute, `Foo` is added for v1.12, then the Attribute is first set:

```
private abstract class AWSSemanticConventionsBase
{
   public virtual string FooAttribute => string.Empty;
}
```

and then defined in 

```
private class AWSSemanticConventions_v1_12 : AWSSemanticConventions_v1_11
{
   public override string FooAttribute => "foo";
}
```